components/zeekopenbsm/src/openbsmconsumer.cpp

```diff
@@ -0,0 +1,363 @@
+    if (tok.tt.hdr32.e_type == AUE_CONNECT) {
+      event.type = Event::Type::Connect;
+    } else if (tok.tt.hdr32.e_type == AUE_BIND) {
+      event.type = Event::Type::Bind;
+    }
+    break;
+  case AUT_HEADER64:
+    event.header.timestamp = tok.tt.hdr64.s;
+    if (tok.tt.hdr64.e_type == AUE_CONNECT) {
+      event.type = Event::Type::Connect;
+    } else if (tok.tt.hdr64.e_type == AUE_BIND) {
+      event.type = Event::Type::Bind;
+    }
+    break;
+  case AUT_HEADER32_EX:
+    event.header.timestamp = tok.tt.hdr32_ex.s;
+    if (tok.tt.hdr32_ex.e_type == AUE_CONNECT) {
+      event.type = Event::Type::Connect;
+    } else if (tok.tt.hdr32_ex.e_type == AUE_BIND) {
+      event.type = Event::Type::Bind;
+    }
+    break;
+  case AUT_HEADER64_EX:
+    event.header.timestamp = tok.tt.hdr64_ex.s;
+    if (tok.tt.hdr64_ex.e_type == AUE_CONNECT) {
+      event.type = Event::Type::Connect;
+    } else if (tok.tt.hdr64_ex.e_type == AUE_BIND) {
+      event.type = Event::Type::Bind;
+    }
+    break;
+  }
+}
+
+void OpenbsmConsumer::extractSubject(IOpenbsmConsumer::Event &event,
+                                     tokenstr_t tok) {
+
+  switch (tok.id) {
+  case AUT_SUBJECT32: {
+    uint32_t pid = tok.tt.subj32.pid;
+    event.header.process_id = pid;
+    event.header.path = getPathFromPid(pid);
+    event.header.user_id = tok.tt.subj32.euid;
+    event.header.group_id = tok.tt.subj32.egid;
+    break;
+  }
+  case AUT_SUBJECT64: {
+    uint32_t pid = tok.tt.subj64.pid;
+    event.header.process_id = pid;
+    event.header.path = getPathFromPid(pid);
+    event.header.user_id = tok.tt.subj64.euid;
+    event.header.group_id = tok.tt.subj64.egid;
+    break;
+  }
+  case AUT_SUBJECT32_EX: {
+    uint32_t pid = tok.tt.subj32_ex.pid;
+    event.header.process_id = pid;
+    event.header.path = getPathFromPid(pid);
+    event.header.user_id = tok.tt.subj32_ex.euid;
+    event.header.group_id = tok.tt.subj32_ex.egid;
+    break;
+  }
+  case AUT_SUBJECT64_EX: {
+    uint32_t pid = tok.tt.subj64_ex.pid;
+    event.header.process_id = pid;
+    event.header.path = getPathFromPid(pid);
+    event.header.user_id = tok.tt.subj64_ex.euid;
+    event.header.group_id = tok.tt.subj64_ex.egid;
+    break;
+  }
+  }
+}
+
+void OpenbsmConsumer::extractReturn(IOpenbsmConsumer::Event &event,
+                                    tokenstr_t tok) {
+
+  switch (tok.id) {
+  case AUT_RETURN32: {
+    int error = 0;
+    if (au_bsm_to_errno(tok.tt.ret32.status, &error) == 0) {
+      if (error == 0) {
+        event.header.success = 1;
+      } else {
+        event.header.success = 0;
+      }
+    } else
+      event.header.success = 0;
+    break;
+  }
+  case AUT_RETURN64: {
+    int error = 0;
+    if (au_bsm_to_errno(tok.tt.ret64.err, &error) == 0) {
+      if (error == 0) {
+        event.header.success = 1;
+      } else {
+        event.header.success = 0;
+      }
+    } else
+      event.header.success = 0;
+    break;
+  }
+  }
+}
```

Comment:
We could make this slightly less repetitive by extracting the status to expect in the switch,
```suggestion

void OpenbsmConsumer::extractReturn(IOpenbsmConsumer::Event &event,
                                    tokenstr_t tok) {
  u_char status = 0;
  switch (tok.id) {
  case AUT_RETURN32:
    status = tok.tt.ret32.status;
    break;
  case AUT_RETURN64:
    status = tok.tt.ret64.err;
    break;
  default:
    return;
  }

  int error = 0;
  if (au_bsm_to_errno(status, &error) == 0) {
    if (error == 0) {
      event.header.success = 1;
    } else {
      event.header.success = 0;
    }
  } else
    event.header.success = 0;
}

```